File: KKY_Tool_Revit_2019-2023/Services/HubCommonOptionsStorageService.vb
Option Explicit On
Option Strict On

Imports System
Imports System.IO
Imports System.Web.Script.Serialization

Namespace Services
    Public NotInheritable Class HubCommonOptionsStorageService
        Private Sub New()
        End Sub

        Public Class HubCommonOptions
            Public Property ExtraParamsText As String = String.Empty
            Public Property TargetFilterText As String = String.Empty
            Public Property ExcludeEndDummy As Boolean
        End Class

        Public Shared Function Load() As HubCommonOptions
            Dim result As New HubCommonOptions()
            Try
                Dim path = GetOptionsPath()
                If Not File.Exists(path) Then
                    Return result
                End If
                Dim json = File.ReadAllText(path)
                If String.IsNullOrWhiteSpace(json) Then
                    Return result
                End If
                Dim serializer As New JavaScriptSerializer()
                Dim loaded = serializer.Deserialize(Of HubCommonOptions)(json)
                If loaded IsNot Nothing Then
                    result = loaded
                End If
            Catch
                Return result
            End Try
            Return result
        End Function

        Public Shared Function Save(options As HubCommonOptions) As Boolean
            If options Is Nothing Then Return False
            Try
                Dim path = GetOptionsPath()
                Dim dir = IO.Path.GetDirectoryName(path)
                If Not String.IsNullOrWhiteSpace(dir) AndAlso Not Directory.Exists(dir) Then
                    Directory.CreateDirectory(dir)
                End If
                Dim serializer As New JavaScriptSerializer()
                Dim json = serializer.Serialize(options)
                File.WriteAllText(path, json)
                Return True
            Catch
                Return False
            End Try
        End Function

        Private Shared Function GetOptionsPath() As String
            Dim root = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
            If String.IsNullOrWhiteSpace(root) Then
                root = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)
            End If
            Dim dir = Path.Combine(root, "KKY_Tool_Revit")
            Return Path.Combine(dir, "hub-common-options.json")
        End Function
    End Class
End Namespace

File: KKY_Tool_Revit_2019-2023/Services/HubCommonOptionsStorageService.cs
using System;
using System.IO;
using System.Web.Script.Serialization;

namespace KKY_Tool_Revit.Services
{
    public static class HubCommonOptionsStorageService
    {
        public sealed class HubCommonOptions
        {
            public string ExtraParamsText { get; set; } = string.Empty;
            public string TargetFilterText { get; set; } = string.Empty;
            public bool ExcludeEndDummy { get; set; }
        }

        public static HubCommonOptions Load()
        {
            var result = new HubCommonOptions();
            try
            {
                var path = GetOptionsPath();
                if (!File.Exists(path)) return result;

                var json = File.ReadAllText(path);
                if (string.IsNullOrWhiteSpace(json)) return result;

                var serializer = new JavaScriptSerializer();
                var loaded = serializer.Deserialize<HubCommonOptions>(json);
                if (loaded != null)
                {
                    result = loaded;
                }
            }
            catch
            {
                return result;
            }

            return result;
        }

        public static bool Save(HubCommonOptions options)
        {
            if (options == null) return false;
            try
            {
                var path = GetOptionsPath();
                var dir = Path.GetDirectoryName(path);
                if (!string.IsNullOrWhiteSpace(dir) && !Directory.Exists(dir))
                {
                    Directory.CreateDirectory(dir);
                }

                var serializer = new JavaScriptSerializer();
                var json = serializer.Serialize(options);
                File.WriteAllText(path, json);
                return true;
            }
            catch
            {
                return false;
            }
        }

        private static string GetOptionsPath()
        {
            var root = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
            if (string.IsNullOrWhiteSpace(root))
            {
                root = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
            }

            var dir = Path.Combine(root, "KKY_Tool_Revit");
            return Path.Combine(dir, "hub-common-options.json");
        }
    }
}
